compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -1204,7 +1242,28 @@ impl<'a> Context<'a> {
             !res_typ.contains_slice_element(),
             "ICE: Nested slice result found during ACIR generation"
         );
-        let value = self.array_get_value(&res_typ, block_id, &mut var_index)?;
+        let mut value = self.array_get_value(&res_typ, block_id, &mut var_index)?;
+
+        if let AcirValue::Var(value_var, typ) = &value {
+            let array_id = dfg.resolve(array_id);
+            let array_typ = dfg.type_of_value(array_id);
+            if let Type::Numeric(numeric_type) = array_typ.first() {
+                if let AcirType::NumericType(num) = typ {
```

Comment:
```suggestion
            if let (Type::Numeric(numeric_type), AcirType::NumericType(num)) = (array_typ.first(), typ) {
```
We should be able to combine these if statements with something like this

Reply:
Done